File: frontend/src/lib/api/cleaning.ts
import { apiClient } from "@/lib/api/client";

export interface CleaningRequest {
  operation_type: string;
  column_name?: string | null;
  params?: Record<string, unknown>;
}

export interface CleaningResult {
  new_version_id: string;
  version_number: number;
  row_count: number;
  column_count: number;
  valid_count: number;
  invalid_count: number;
}

export function applyCleaning(datasetId: string, payload: CleaningRequest) {
  return apiClient.post<CleaningResult>(`/api/datasets/${datasetId}/clean`, payload);
}

File: backend/app/schemas/cleaning.py
import uuid

from pydantic import BaseModel


class CleaningRequest(BaseModel):
    operation_type: str
    column_name: str | None = None
    params: dict = {}


class CleaningResponse(BaseModel):
    new_version_id: uuid.UUID
    version_number: int
    row_count: int
    column_count: int
    valid_count: int
    invalid_count: int

    model_config = {"from_attributes": True}
